libtransmission/resume.cc

```diff
@@ -590,24 +569,13 @@ auto loadProgress(tr_variant* dict, tr_torrent* tor, tr_torrent::ResumeHelper& h
                 rawToBitfield(blocks, buf, buflen);
             }
         }
-        else if (auto sv = std::string_view{}; tr_variantDictFindStrView(prog, TR_KEY_have, &sv))
-        {
-            if (sv == "all"sv)
-            {
-                blocks.set_has_all();
-            }
-            else
-            {
-                err = "Invalid value for HAVE";
-            }
-        }
```

Comment:
Will removing this break `.resume` files that were written to disk before this PR?

Reply:
I don't think it will.

No information is lost even if we discard the `have` key. It was added in 4917cd119b94e128805d8ccbc0b0c2b980f2139c to speed up the `bitfield` key (now replaced by the `blocks` key), but the `bitfield`/`blocks` key was always written to the resume file no matter if the `have` key is applicable or not.

On top of that, from 4917cd119b94e128805d8ccbc0b0c2b980f2139c onwards, the `have` key is completely redundant. Firstly, the `blocks` key has subsumed the functionality of the `have` key. And secondly, when reading the resume file, the `blocks` key is considered before the `have` key in an `else-if` chain, so the `have` key basically had no effect since then.